ui/app/src/modules/Preview/ToolsPanel.tsx

```diff
@@ -60,15 +63,25 @@ export default function ToolsPanel() {
   const dispatch = useDispatch();
   const site = useActiveSiteId();
   const classes = useStyles();
-  const { showToolsPanel } = usePreviewState();
+  const { showToolsPanel, toolsPanel } = usePreviewState();
   const toolsPanelWidth = useSelection<number>((state) => state.preview.toolsPanelWidth);
   const pages = useSelection<WidgetDescriptor[]>((state) => state.preview.toolsPanelPageStack);
   const uiConfig = useSiteUIConfig();
   const baseUrl = useSelection<string>((state) => state.env.authoringBase);
+  const { username } = useActiveUser();
 
-  const resource = useLogicResource<WidgetDescriptor[], WidgetDescriptor[]>(uiConfig.preview.toolsPanel.widgets, {
+  useEffect(() => {
+    if (nnou(uiConfig.xml) && !toolsPanel) {
+      const storedPage = getStoredPreviewToolsPanelPage(site, username);
+      dispatch(
+        initToolsPanelConfig({ configXml: uiConfig.xml, references: uiConfig.references, pageStack: storedPage })
```

Comment:
Why send references?
```suggestion
        initToolsPanelConfig({ configXml: uiConfig.xml, storedPageStack: storedPage })
```